lineax/_misc.py

```diff
@@ -19,6 +19,7 @@
 import jax.core
 import jax.numpy as jnp
 import jax.tree_util as jtu
+from jax import ShapeDtypeStruct
```

Comment:
Nit: I'd just `jax.ShapeDtypeStruct` directly for consistency with the rest of our imports.